### DISCUSSION TITLE
Google Search Engine

### DISCUSSION THREAD

**Discussion by sudo-nautilus** [Eloston/ungoogled-chromium#1488] [Q&A] [answered]
This may be 'crazy' but I need google as my search engine
plus autosuggestions aren't working properly

**Comment by tangalbert919** [comment]
You can try going to their website and using their search engine from there.

**Reply by sudo-nautilus** [reply]
I can't do that,
I'm too lazy 😂

**Reply by TJ-WEB-EXPLORER** [reply]
oh

**Comment by Ahrotahn** [comment]
You might be able to add it as a custom search engine.  

You can get there from settings > manage search engines.  Or `chrome://settings/searchEngines`.  
Then add `https://google.com/search?q=%s` as the URL.

**Reply by sudo-nautilus** [reply]
```
http://google.com/search?q=%s&{google:RLZ}{google:originalQueryForSuggestion}{google:assistedQueryStats}{google:searchFieldtrialParameter}{google:iOSSearchLanguage}{google:prefetchSource}{google:searchClient}{google:sourceId}{google:contextualSearchVersion}ie={inputEncoding}
```

This working for me, anyway

**Reply by Klipperken** [reply]
@sudo-nautilus @Ahrotahn 
I did this in my Edge, however ... it does not accept it.
That is:
I have "&num=100" at the end, but I keep getting 10 results, then i need to hit a new page, which triggers my severe Asperger's that is then further fueled in anger by my severe ADHD.  :(
While Google Search goes as normal ... the "&num=100" is not activated.

Strange, since, if I do a query, hit enter, get the result URL, and add this bit at the end, then hit enter, I DO get 100 results per page.
I fail to get this ... .

How can I fix this, please?

**Reply by tomasz1986** [reply]
What's your URL specifically?

`https://www.google.com/search?q=%s&num=100` works fine here.

**Reply by Klipperken** [reply]
@tomasz1986 
Thank you for responding.

I used https://www.google.com/search?q=%s&{google:RLZ}{google:originalQueryForSuggestion}{google:assistedQueryStats}{google:searchFieldtrialParameter}{google:language}{google:prefetchSource}{google:searchClient}{google:sourceId}{google:contextualSearchVersion}ie={inputEncoding}&num=100 ,
sir.
The same part I have in the settings: {google:baseURL}search?q=%s&{google:RLZ}{google:originalQueryForSuggestion}{google:assistedQueryStats}{google:searchFieldtrialParameter}{google:language}{google:prefetchSource}{google:searchClient}{google:sourceId}{google:contextualSearchVersion}ie={inputEncoding}&num=100 .
I changed the {google:baseURL} part with https://www.google.com/ .
Probably too much to get a result, as I kept getting a 404?

What is this bit called, by the way, in terminology?

I'll try yours, good sir.

**Reply by Klipperken** [reply]
@tomasz1986 
Hilarious:
It gives me a search result for %s instead of just the base Google page with 100 results?

**Reply by tomasz1986** [reply]
![image](https://github.com/user-attachments/assets/1397dcf1-f691-4e73-b70d-373385a4cb51)

**Reply by Klipperken** [reply]
Yes, well, as I said earlier, I have made a new search engine with the string I gave.
No issues, works neatly ... except no 100 results.

So, I want to make a bookmark which gives 100 results/page.
How do I do this without stumbling on the %s search result when opening said bookmark?

**Reply by tomasz1986** [reply]
`https://www.google.com/?num=100`.

**Reply by Klipperken** [reply]
10 results per page ... .  :'(

**Reply by tomasz1986** [reply]
I think some screenshots would help to understand what you're doing. The above link used as a bookmark does work fine here for me, i.e. the main google site opens, and if you search something, the `?num=100` part stays in the URL, making Google display more results.

**Reply by Klipperken** [reply]
Settings: Manage Search Engines:
![Settings Manage Search Engines](https://github.com/user-attachments/assets/2f4eef92-e1bd-453d-b910-842466f2ac95)
Settings: Address Bar Seach Engines:
![Settings Address Bar Search Engines](https://github.com/user-attachments/assets/714b77b1-6193-4de5-8fab-c202abaf8a4c)

The bookmark:
![Google Bookmark](https://github.com/user-attachments/assets/15de282d-d33f-41cd-8447-e901fd41e9bf)

Google result:
![Google Result](https://github.com/user-attachments/assets/d0b08beb-8cc2-47a8-a094-a2ea7273b704)
10 results per page

**Reply by Klipperken** [reply]
@tomasz1986 
Hope this helps, sir?

**Reply by Klipperken** [reply]
Oh, wait:
When I make a query and enter, I get the results, 10 per page.
If I add &num=100 at the end of the search result and again enter, it works perfectly.

However, I do not want to add this each new search, sir.
I am looking to fire up google search with 100 results ready, or, automated, if you like.

**Reply by tomasz1986** [reply]
Something seems up. What happens if you just go to https://www.google.com/?num=100 directly and search there? Please test in Incognito mode or a clean profile to make sure that nothing else has been interfering.

**Reply by Klipperken** [reply]
Left-clicked the link:
10 results/page. sir.

Edit:
Right-click, open in new tab:
Ditto.

**Reply by Klipperken** [reply]
OH!!!
In my Firefox ... it works perfect.
What?
Why not in my Edge?  o_O

**Reply by tomasz1986** [reply]
Yeah, so this is the problem. For me, it displays the requested number of results per page here. I've also just tested this in a clean profile, in plain Chromium, and even in Brave, and the behaviour is the same in all of them.

Can you see what happens in any other browser? Also, where does your Ungoogled Chromium installation come from?

**Reply by Klipperken** [reply]
Yes, last post of mine, did this.

**Reply by tomasz1986** [reply]
> Why not in my Edge? o_O

This is a support forum for Ungoogled Chromium 😉. No idea what Edge is doing differently.

**Reply by Klipperken** [reply]
Both should accept this link, equally.
This is ... baffling!!!

Thank you, sir, and apologies for keeping you so busy ... if I had tested this sooner on my Firefox, we would not have wasted time so long.
Again, sincere apologies.

**Comment by sudo-nautilus** [comment]
Thank you for replying. 
That things works except one thing - search autosuggestions

**Comment by sudo-nautilus** [comment]
`Suggestions URL with %s in place of query`
\- What to do with that, I guess it is the thing which triggers search autosuggestions

**Comment by sudo-nautilus** [comment, marked as answer]
To get Google Search Engine, Do the Following:
- Go to chrome://settings/searchEngines
- Click `Add` Button near `Other search engines`
- Add `Google` In `Search Engine` input box
- Add `google.com` in `Keyword` input box
- Add `https://www.google.com/search?q=%s&{google:RLZ}{google:originalQueryForSuggestion}{google:assistedQueryStats}{google:searchFieldtrialParameter}{google:iOSSearchLanguage}{google:prefetchSource}{google:searchClient}{google:sourceId}{google:contextualSearchVersion}ie={inputEncoding}` in `URL with %s in place of query` input box
- Add `https://www.google.com/complete/search?client=chrome&q=%s` in `Suggestions URL with %s in place of query` input box
- Click The Blue `Save` Button
- Click the three dot menu near `Google` and select `Make default`

Here's a brief screenshot of the final result :

![screenshot](https://user-images.githubusercontent.com/78522210/114988160-a33dc100-9eb3-11eb-807a-9b1e4459d8d6.png)

**Reply by notsidney** [reply]
Hey, are there any other settings you have on? I tried this on v95.0.4638.54 and it doesn’t work for me :/

**Reply by networkException** [reply]
What exactly does not work for you?

**Reply by faelana** [reply]
> Hey, are there any other settings you have on? I tried this on v95.0.4638.54 and it doesn’t work for me :/

Try to follow my instructions. What doesnt work?

**Reply by notsidney** [reply]
Oh, sorry! I realized I didn’t put enough context in that comment. Google search works for me, but search suggestions don’t appear in the omnibox. I’ve tried the suggestion URL `https://www.google.com/complete/search?client=chrome&q=%s` and also without `client=chrome&`. I also tried the search URL above and `https://www.google.com/search?q=%s` without the other tracking parameters. I also restarted Chromium entirely, but none of that has worked.

**Reply by PF4Public** [reply]
We should probably put this into the FAQ?

**Reply by networkException** [reply]
Sure why not

**Reply by Moon1moon** [reply]
> notsidney

did you find any solution? As you said search suggestions don’t appear in the omnibox.

**Reply by notsidney** [reply]
No, it still only shows suggestions from my history, and not Google search suggestions.

**Reply by Moon1moon** [reply]
> No, it still only shows suggestions from my history, and not Google search suggestions.

I found today.

1. Go to `chrome://settings/searchEngines`

_If you have old search engine related google, delete it before the these settings._

2. Find Other search engines and click to Add button

3. Fill form like this:

   **Search engine:**
   `Google`
**Keyword:**
`Google.com`
**URL with %s in place of query:**
`https://www.google.com/search?q=%s`
**Suggestions URL with %s in place of query:**
`https://www.google.com/complete/search?client=chrome&q=%s`
4. Click `save `button,
5. Click the `three dot` menu near `Google` and select `Make default`
6. Go to `chrome://settings/syncSetup`
7. `Enable `the` Autocomplete searches and URLs option`

Now you can use search suggestions.

**Reply by notsidney** [reply]
> 6. Go to `chrome://settings/syncSetup`
> 7. `Enable `the` Autocomplete searches and URLs option`

This was what I was missing! Thanks.

**Comment by faelana** [comment]
i have a problem with this, when i do a search other strings are included in the url, one shows a past search that i made and all the others look to be encoded. is there a way to make this any better?
@sudo-nautilus @Ahrotahn

**Reply by Ahrotahn** [reply]
Do you mean that they show up as suggestions, or that it's on the same line you're inputting text on?

If it's the former, then you might be interested in the "Omnibox Autocomplete Filtering" flag.  You can find that in `chrome://flags` or use the switch documented on [docs/flags.md](https://github.com/Eloston/ungoogled-chromium/blob/master/docs/flags.md).

**Reply by faelana** [reply]
@Ahrotahn i have suggestions disabled, i mean on the url after searching it.

**Reply by networkException** [reply]
Instead of including all the tracking parameters Google uses you can also just use `https://google.com/search?q=%s`

**Reply by faelana** [reply]
> Instead of including all the tracking parameters Google uses you can also just use `https://google.com/search?q=%s`

@networkException i tried that but it seems to always ignore it and input the tracking parameters..

**Reply by faelana** [reply]
Okay, i just fixed it. 
To remove the tracking parameters do the following while using google as a search engine:
1º - First remove the alredy generated google entry. (editing doesnt work, it overwrites _https://google.com/search?q=%s_ with the tracking parameters.)
2º - Use the "Add" button to create a new search engine. I setup mine like this:
Search Engine: Google
Keyword: google.com
URL with %s in place of query: https://google.com/search?q=%s
After that it will work as expected without the tracking parameters.

**Comment by Zerogoki00** [comment]
@sudo-nautilus I opened search engine settings but there is no "Other search engines". There are only "Search engines" , "Site search", "Inactive shortcuts". What should I do?
![image](https://user-images.githubusercontent.com/61648511/217195975-5e540dcd-bce1-443b-ae0e-22247c1f0698.png)

**Reply by networkException** [reply]
I think that has merely been renamed to "Site search"

**Reply by Zerogoki00** [reply]
yes, thanks

**Comment by leros1337** [comment]
Still cant add search engine and make it default. No button, site search != search engine

**Comment by ContactRecipient** [comment]
having the same issue. still unable to set google as my default search engine, let alone adding a custom search engine. The "Add" button is simply not there.

**Reply by networkException** [reply]
What build of ungoogled-chromium are you using?

**Reply by ContactRecipient** [reply]
109.0.5414.120

**Reply by networkException** [reply]
First of all you should update, that version is highly outdated and vulnerable. Then I also meant what platform, where you go the binary from, etc. I don't have any issues on archlinux which means this is unlikely an issue with the patchset but rather with packaging

**Reply by ContactRecipient** [reply]
im on windows. just switched to version 111.0.5563.65 and the issue persists.

**Reply by leros1337** [reply]
110.0.5481.100 (Official Build, ungoogled-chromium) - macos installed via brew

**Reply by networkException** [reply]
@ContactRecipient are you on a managed install by any chance? I'm on a (pretty much) fresh install of 111.0.5563.65 on Windows and cannot reproduce the issue.

As for macOS I don't have an install of that at hand directly right now, could you (both perhaps) check if using an empty profile helps?

![image](https://user-images.githubusercontent.com/42888162/224846795-58c3248d-2685-4919-8c72-e73e65ab2d69.png)

**Reply by dsvi** [reply]
You only have `Add` button in the site search section. Not at search engines

**Comment by simondemeule** [comment]
I encountered the same issue today on macOS, on both an older version and on 110.0.5481.177 (the most recent version available through brew at the moment). 

After trying out things semi-randomly, I've found a solution. The UI is just very confusing. You need to first add an entry to "Site search" with a temporary shortcut, then click the dotted icon, and promote it to default using the dropdown menu that pops up. The entry is then moved to "Search engines" section.

Maybe change the UI so there is also an "Add" button on the "Search engines" section? The fact there is an "Add" button on the "Site search" section but not on the "Search engines" section, and the fact that UI element that provides this feature is hidden implicitly tells the user that search engines cannot be added. This seems to have confused enough people over the past two years for a dozen tech-literate people to care enough to show up here and puzzle over this.

**Reply by leros1337** [reply]
worked for me!
1) google
2) google.nl (somewhat .com not allowed, use any other)
3) https://www.google.com/search?q=%s
4) https://www.google.com/complete/search?client=chrome&q=%s

add > 3 dot > make default > go back to main left bar > click search > change to google

**Reply by gamma19** [reply]
Thanks alot! After searching all of the internet, this is what I was looking for !

**Reply by aabedraba** [reply]
If someone's coming in 2025, this is the thread that gives you the solution!

**Reply by Ceraviews** [reply]
🐐

**Reply by MusaGillani** [reply]
🐐

**Comment by itsnikolayy** [comment]
For anyone still having this issue like I did, this is how I figured out how to add a custom search engine (Google) to Ungoogled Chrome:
### 1. Navigate to `chrome://settings/searchEngines`
<img width="1871" height="1079" alt="image" src="https://github.com/user-attachments/assets/699c50e8-db9c-4f92-9c33-d4508a95bd21" />


### 2. Scroll down a little bit to `Site search` and click the `Add` button
<img width="725" height="396" alt="image" src="https://github.com/user-attachments/assets/b1a237bf-bfa1-4336-bbfa-37cd8ee49818" />

### 3. Fill out the `Add site search` box, then click `Add` when filled out
For example, here is how you add Google:
Name:
`Google`
Shortcut:
`google.com`
URL with %s in place of query:
`https://www.google.com/search?q=%s`
Suggestions URL with %s in place of query:
`https://www.google.com/search?q=%s&{google:RLZ}{google:originalQueryForSuggestion}{google:assistedQueryStats}{google:searchFieldtrialParameter}{google:iOSSearchLanguage}{google:prefetchSource}{google:searchClient}{google:sourceId}{google:contextualSearchVersion}ie={inputEncoding}`
<img width="694" height="658" alt="image" src="https://github.com/user-attachments/assets/4d7cfa3d-9130-4c8f-85ce-e43b8dc4f516" />

### 4. Finally, click the `...` 3 dots of the search engine you added, and click `Make default`. Done!
<img width="712" height="369" alt="image" src="https://github.com/user-attachments/assets/4407bcc0-f0e0-46c7-ba15-357e3469597f" />


The good news is, this is a one time setting change. I had Ungoogled chrome 119 up until today (12/16/2023 - 8/13/2025) when sites finally started breaking.

I am using Ungoogled chrome 138.0.7204.183-1 for Windows https://ungoogled-software.github.io/ungoogled-chromium-binaries/releases/windows/64bit/

Additionally, if you're unable to use manifest v2 extensions (uBlock Origin etc.) with your version of chrome, use a chrome/chromium version 138 or older, and v2 manifest extensions will still work!

**Reply by tomasz1986** [reply]
Is there any particular reason why you're using all those search parameters in the suggestions URL? Normally, something like

```
https://www.google.com/complete/search?client=chrome&q=%s
```

is enough for the suggestions to work.

**Comment by aficus11** [comment]
I can't fucking believe it. Just fucking add Google in Search engines list, morons.

Or I suppose to do that with all my 100 profiles? Honestly, so retarded. 

I get removing Google tracking from browser. But this is a fucking url search query. Just add it, idiots!!!
